### PR TITLE
feat: differentiate Java version build

### PR DESCRIPTION
**Pull request by guangie88**
feat: differentiate Java version build
    
Pin to Buster build by overriding Dockerfiles.
    
Build for only latest x.y versions, from 2.4 to 3.1.

### CODE REVIEW

File: overrides/base/2.4.z/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+COPY ${img_path}/spark/entrypoint.sh /opt/
+COPY examples /opt/spark/examples
+COPY ${k8s_tests} /opt/spark/tests
+COPY data /opt/spark/data
+
+ENV SPARK_HOME /opt/spark
+
+WORKDIR /opt/spark/work-dir
+
+ENTRYPOINT [ "/opt/entrypoint.sh" ]
```
Comment (guangie88) [inline overrides/base/2.4.z/Dockerfile:58]: https://github.com/apache/spark/blob/v2.4.8/resource-managers/kubernetes/docker/src/main/dockerfiles/spark/Dockerfile

Adapted from above, and changed the base image to allow override for `-buster` build (current build uses the later `bullseye` due to upgrade to `openjdk` images, which does not have Python 2.7 and uses Python 3.9, which are both issues).

File: overrides/python/2.4.z/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    pip install --upgrade pip setuptools && \
+    # You may install with python3 packages by using pip3.6
+    # Removed the .cache to save space
+    rm -r /root/.cache && rm -rf /var/cache/apt/*
+
+COPY python/pyspark ${SPARK_HOME}/python/pyspark
+COPY python/lib ${SPARK_HOME}/python/lib
+
+WORKDIR /opt/spark/work-dir
+ENTRYPOINT [ "/opt/entrypoint.sh" ]
```
Comment (guangie88) [inline overrides/python/2.4.z/Dockerfile:37]: https://github.com/apache/spark/blob/v2.4.8/resource-managers/kubernetes/docker/src/main/dockerfiles/spark/bindings/python/Dockerfile

Similar, except remove `PYTHONENV`, and adapted to do `COPY python/pyspark ${SPARK_HOME}/python/pyspark` from:
https://github.com/apache/spark/blob/v3.0.3/resource-managers/kubernetes/docker/src/main/dockerfiles/spark/bindings/python/Dockerfile#L39

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # CHANGELOG
 
+## v3
+
+- (Temporarily drop support for R due to keyserver issues)
```
Comment (xtrntr) [inline CHANGELOG.md:5]: 👍

File: templates/vars.yml [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
 versions:
-- spark:  ['2.4.7']
+- spark:  ['2.4.8']
+  java:   ['8']
   hadoop: ['2.7.3', '3.1.0']
   scala:  ['2.11', '2.12']
 
-- spark:  ['3.0.0', '3.0.1', '3.0.2', '3.1.1']
+- spark:  ['3.0.3', '3.1.2']
+  java:   ['8', '11']
```
Comment (xtrntr) [inline templates/vars.yml:10]: Do we need to support Java 8 for Spark 3, or is this just a nice to have?

**Review by xtrntr — APPROVED** [review]

**Pull request merged**